SecurePayAU: Send order ID for payments with stored card

dacook opened the pull request:
**Changelog:** SecurePayAU: Send order ID for payments with stored cards

Order IDs were being sent for normal payments, but now they are also sent for payments triggered on stored cards as `transactionReference`.
As per XML spec (https://auspost.com.au/payments/docs/securepay/resources/API_Card_Storage_and_Scheduled_Payments.pdf )

Also:
- Fixed broken remote tests
- Send content-type headers for best practice

Unit test results:
test/unit/gateways/secure_pay_au_test.rb
24 tests, 106 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote test results: 
test/remote/gateways/remote_secure_pay_au_test.rb
18 tests, 57 assertions, 0 failures, 0 errors, 0 pendings, 2 omissions, 0 notifications
100% passed

dacook commented:
It would be great if someone could take a look at this. 
We have been using this code in production for a long time.

dacook commented:
@rachelkirk and @curiousepic, apologies for the ping, but I'm not sure what the process is for reviewing PRs and I see that you've recently approved a PR. 

We've been using this code since 2018, and submitted the PR in 2021 as per the contributing guide. We've also submitted another PR [here](https://github.com/activemerchant/active_merchant/pull/4001).

Would you or somebody be able to take a look? We'd love to be able to use and further support the official gem, rather than our own fork.

dacook commented:
PS I actually have some spec improvements to offer too, but as they depend on this PR I never bothered submitting them..

dacook commented:
Hi @aenand I'm pinging you as you have recently reviewed some PRs. This is due to be closed, however I believe it has value and is ready to go (including specs as per the guide). Can you please take a look or assign to the relevant person?

> We've been using this code since 2018, and submitted the PR in 2021 as per the contributing guide. We've also submitted another PR [here](https://github.com/activemerchant/active_merchant/pull/4001).

aenand commented:
> Hi @aenand I'm pinging you as you have recently reviewed some PRs. This is due to be closed, however I believe it has value and is ready to go (including specs as per the guide). Can you please take a look or assign to the relevant person?
> 
> > We've been using this code since 2018, and submitted the PR in 2021 as per the contributing guide. We've also submitted another PR [here](https://github.com/activemerchant/active_merchant/pull/4001).
> 
> PS let me know if you'd like me to rebase it.

Hey @dacook I'm so sorry this has slipped through the cracks. If you rebase this and add a changelog I will look to merge this

dacook commented:
Thanks. It looks like the PR name should be enough for the changelog, and in fact [the wiki states](https://github.com/activemerchant/active_merchant/wiki/Submitting-Your-Gateway#best-practices):

> Does not update the CHANGELOG

Should I add the PR name to the CHANGELOG file in this PR, and if so can you please update the wiki to make that clearer?

dacook commented:
I'm unable to install gems for the new ruby version 3.1:

```
Resolving dependencies...
nokogiri-1.11.7-x86_64-darwin requires ruby version < 3.1.dev, >= 2.5, which is incompatible with the current version,
ruby 3.1.4p223
```

I tried using Ruby 2.7.8, but it was also refused:

```
Resolving dependencies...
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    activemerchant was resolved to 1.136.0, which depends on
      ruby (>= 3.1)

Could not find gem 'ruby (>= 3.1)', which is required by gem 'activemerchant', in any of the relevant sources:
  the local ruby installation
```

Therefore I can't re-run the tests until that is fixed.

dacook commented:
@aenand I have rebased and added a changelog line as requested.
Can you please review my above comments also?

dacook commented:
Hi @aenand, can you please review this longstanding PR? 
I have added a CHANGELOG message and rebased _again_. 
It's worth noting that this code has been used in production since 2018.

dacook commented:
Hi @jcreiff, @almalee24, @DustinHaefele, I'm pinging you because you recently approved or merged PRs. I don't understand the review process or why newer PRs seem to be prioritised over older ones like this one. Can you please review and explain if anything is wrong with it? 

It's a simple addition that has been used in production since 2018.

The CHANGELOG message that was requested is obviously conflicting. I'd be happy to rebase and update that one more time if someone will actually look at it this time.

dacook commented:
sigh 🙄

DustinHaefele reviewed (approved):
This looks good. I ran all the tests locally and feel good about merging it.

DustinHaefele commented:
Hey @dacook, I hate to ask one more thing from you I know we have failed to get this in for you for years now, which is ridiculous. But can you rebase one more time. There is a merge conflict, though it is just on the changelog. I was going to resolve, but didn't want to push to your repo.

I know our timezones are all off, but I have left a note on my desk to look at this and merge first thing tomorrow the morning so that you have waking hours on your side of the globe to do the rebase.

dacook commented:
Hi @DustinHaefele , thanks for following up. Sorry I didn't see this earlier. I've rebased to fix the changelog conflict (and also updated the URL in the last commit message), so it should be ready to go!
